Utils/Extensions/StringExtensions.swift

```diff
@@ -52,6 +52,31 @@ public extension String {
         return str as! String
     }
 
+    /**
+    Truncates a String in the middle only if it's longer than `maxLength`
```

Comment:
I notice that this behaves incorrectly when maxLength is odd. That is, I expect the following equation to always be true:

```
  count(longString.truncate(maxLength: x)) == x +  1
```

(assuming … not ...), but it's not true.